src/System.Diagnostics.Debug/tests/DebugTestsNoListeners.cs

```diff
@@ -0,0 +1,235 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    // These tests test the static Debug class. They cannot be run in parallel
+    // DebugTestsNoListeners: when run alone, tests Debug behavior before Debug is set with Trace Listeners.
```

Comment:
What does guarantee that DebugTestsNoListeners run first before DebugTestsUsingListeners ?

Reply:
I don't know if there is any guarantee. I am open to suggestions.

Reply:
Maybe reset the DebugProvider static field in corelib to its initial value?

Reply:
This is how I made sure each test class uses the correct DebugProvider: https://github.com/dotnet/corefx/pull/33020/commits/a4ec999b288a3863dd001aaa6cf342a0a85d87fb